improver_tests/calibration/simple_bias_correction/test_ApplyBiasCorrection.py

```diff
@@ -0,0 +1,289 @@
+    bias_cube = generate_bias_cubelist(
+        num_bias_inputs, single_frt_with_bounds=True, masked_data=masked_bias_data
+    )[0]
+
+    expected = TEST_FCST_DATA - MEAN_BIAS_DATA
+    if fill_masked_bias_data and masked_bias_data:
+        expected = np.where(MASK, TEST_FCST_DATA, expected)
+
+    result = apply_additive_correction(forecast_cube, bias_cube, fill_masked_bias_data)
+
+    if masked_bias_data and not fill_masked_bias_data:
+        assert isinstance(result, ma.masked_array)
+        assert np.all(result.mask == MASK)
+        assert np.ma.allclose(result, expected, atol=0.05)
+    else:
+        assert isinstance(result, np.ndarray)
+        assert np.allclose(result, expected, atol=0.05)
+
+
+def test__init__():
+    """Test that the class functions are set to the expected values."""
+    plugin = ApplyBiasCorrection()
+    assert plugin.correction_method == apply_additive_correction
+
+
+@pytest.mark.parametrize("single_input_frt", (False, True))
+def test_get_mean_bias(single_input_frt):
+    """Test that mean value cube returned has expected properties."""
+    input_cubelist = generate_bias_cubelist(30, single_frt_with_bounds=single_input_frt)
+    result = ApplyBiasCorrection()._get_mean_bias(input_cubelist)
+
+    # Check that the CubeList has been collapsed down to a single value along
+    # the forecast_reference_time coord.
+    assert "forecast_reference_time" not in get_dim_coord_names(result)
+    # Check that the resultant value is the expected mean value (within tolerance).
+    assert np.allclose(result.data, MEAN_BIAS_DATA, atol=0.05)
+    # Check that the return type is an iris.cube.Cube
+    assert isinstance(result, Cube)
+    # Check conistent datatype
+    assert result.dtype == input_cubelist[0].dtype
+
+
+def test_get_mean_bias_fails_on_inconsistent_bounds():
+    """Test that get_mean_bias fails when passing in multiple bias values defined
+    over a range of forecast_reference_times."""
+    # Set up cube inputs defined over multiple frt values.
+    input_cubelist = CubeList()
+    input_cubelist.extend(generate_bias_cubelist(2, single_frt_with_bounds=True))
+    input_cubelist.extend(
+        generate_bias_cubelist(
+            2,
+            single_frt_with_bounds=True,
+            last_valid_time=VALID_TIME - timedelta(days=2),
+        )
+    )
+    with pytest.raises(ValueError):
+        ApplyBiasCorrection()._get_mean_bias(input_cubelist)
+    # Set up cube inputs with mixed definition: one over single frt, the other over
+    # multiple frt values.
+    input_cubelist = CubeList()
+    input_cubelist.extend(generate_bias_cubelist(2, single_frt_with_bounds=False))
+    input_cubelist.extend(
+        generate_bias_cubelist(
+            2,
+            single_frt_with_bounds=True,
+            last_valid_time=VALID_TIME - timedelta(days=2),
+        )
+    )
+    with pytest.raises(ValueError):
+        ApplyBiasCorrection()._get_mean_bias(input_cubelist)
```

Comment:
Good suggestion. This will key is to test the case where two cubes have bounds associated and where one does and the other doesn't. Both these case can be accommodated with the same code using the `@pytest.mark.parametrize` as suggested.